From 304d6486d0fc2d2267cc8845d637a052c1f6b72a Mon Sep 17 00:00:00 2001
From: KunYi Chen <kunyi.chen@gmail.com>
Date: Thu, 17 Nov 2022 17:36:47 +0800
Subject: [PATCH 1/5] upgrade pyinstaller version to v5.6.2

due to older version on Debian/another linux have a bug.
can't find libpython.so
---
 .../python/pyinstaller_4.00.bb                | 17 -------------
 ...thon3-pyinstaller-hooks-contrib_2022.13.bb | 18 ++++++++++++++
 .../python/python3-pyinstaller_5.6.2.bb       | 24 +++++++++++++++++++
 3 files changed, 42 insertions(+), 17 deletions(-)
 delete mode 100644 meta-ma35d1/recipes-devtools/python/pyinstaller_4.00.bb
 create mode 100644 meta-ma35d1/recipes-devtools/python/python3-pyinstaller-hooks-contrib_2022.13.bb
 create mode 100644 meta-ma35d1/recipes-devtools/python/python3-pyinstaller_5.6.2.bb

diff --git a/meta-ma35d1/recipes-devtools/python/pyinstaller_4.00.bb b/meta-ma35d1/recipes-devtools/python/pyinstaller_4.00.bb
deleted file mode 100644
index c4d41d4..0000000
--- a/meta-ma35d1/recipes-devtools/python/pyinstaller_4.00.bb
+++ /dev/null
@@ -1,17 +0,0 @@
-SUMMARY = " pyinstaller_4.00 "
-
-LICENSE = "BSD"
-LIC_FILES_CHKSUM = "file://COPYING.txt;md5=702f886187082d5370323fae257c8523"
-
-inherit native pypi setuptools3
-
-SRCREV= "19fb799a11d2d796fc8758808f873c40e2bf5118"
-
-SRC_URI = "git://github.com/pyinstaller/pyinstaller.git;protocol=https;branch=master"
-S = "${WORKDIR}/git"
-B =  "${WORKDIR}/build"
-
-BBCLASSEXTEND = "native nativesdk"
-
-INHIBIT_SYSROOT_STRIP = "1"
-INSANK_SKIP_${PN}_append = "already-stripped"
diff --git a/meta-ma35d1/recipes-devtools/python/python3-pyinstaller-hooks-contrib_2022.13.bb b/meta-ma35d1/recipes-devtools/python/python3-pyinstaller-hooks-contrib_2022.13.bb
new file mode 100644
index 0000000..18025fa
--- /dev/null
+++ b/meta-ma35d1/recipes-devtools/python/python3-pyinstaller-hooks-contrib_2022.13.bb
@@ -0,0 +1,18 @@
+SUMMARY = "Community maintained hooks for PyInstaller"
+HOMEPAGE = "https://github.com/pyinstaller/pyinstaller-hooks-contrib"
+AUTHOR = " <>"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=822bee463f4e00ac4478593130e95ccb"
+
+SRCREV = "dfc2087824547461c40b54ff8faf229036737b3b"
+SRC_URI = "git://github.com/pyinstaller/pyinstaller-hooks-contrib.git;protocol=https;branch=master"
+SRC_URI[sha256sum] = "e06d0881e599d94dc39c6ed1917f0ad9b1858a2478b9892faac18bd48bcdc2de"
+
+S = "${WORKDIR}/git"
+B =  "${WORKDIR}/build"
+
+RDEPENDS_${PN} = ""
+
+inherit native pypi setuptools3
+
+BBCLASSEXTEND = "native nativesdk"
diff --git a/meta-ma35d1/recipes-devtools/python/python3-pyinstaller_5.6.2.bb b/meta-ma35d1/recipes-devtools/python/python3-pyinstaller_5.6.2.bb
new file mode 100644
index 0000000..bf6b2a4
--- /dev/null
+++ b/meta-ma35d1/recipes-devtools/python/python3-pyinstaller_5.6.2.bb
@@ -0,0 +1,24 @@
+SUMMARY = "PyInstaller bundles a Python application and all its dependencies into a single package."
+HOMEPAGE = "http://www.pyinstaller.org/"
+AUTHOR = "Hartmut Goebel, Giovanni Bajo, David Vierra, David Cortesi, Martin Zibricky <>"
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://COPYING.txt;md5=6caf9c6da7a5d7bd095b1f2a7b118b8a"
+
+SRCREV = "09b8a1ebd0a62c4e61de61cd33c739c997249a89"
+
+SRC_URI = "git://github.com/pyinstaller/pyinstaller.git;protocol=https;branch=develop"
+SRC_URI[sha256sum] = "865025b6809d777bb0f66d8f8ab50cc97dc3dbe0ff09a1ef1f2fd646432714fc"
+
+S = "${WORKDIR}/git"
+B =  "${WORKDIR}/build"
+
+RDEPENDS_${PN} = "python3-setuptools python3-altgraph python3-pyinstaller-hooks-contrib"
+
+DEPENDS += "python3-wheel-native"
+
+inherit native pypi setuptools3
+
+BBCLASSEXTEND = "native nativesdk"
+
+INHIBIT_SYSROOT_STRIP = "1"
+INSANK_SKIP_${PN}_append = "already-stripped"
-- 
2.38.1

